### ISSUE TITLE
Elixir 1.17.3 test errors in "logger (ex_unit)" on macOS with Erlang/OTP 27.1

### ISSUE THREAD

**Issue by pertsevds** [elixir-lang/elixir#13848]
### Elixir and Erlang/OTP versions

Erlang/OTP 27.1
```shell
# erl -v
Erlang/OTP 27 [erts-15.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]
```

### Operating system

macOS 14.6.1

### Current behavior

Steps to reproduce:
```shell
brew install asdf
asdf plugin add erlang https://github.com/asdf-vm/asdf-erlang.git
asdf install erlang 27.1
asdf global erlang 27.1
asdf plugin-add elixir https://github.com/asdf-vm/asdf-elixir.git
asdf install elixir ref:78f63d08313677a680868685701ae79a2459dcc1 # 1.17.3 tag
```

Error message:
```shell
==> logger (ex_unit)
Running ExUnit with seed: 568247, max_cases: 16

....................................

  1) test translates Supervisor progress (Logger.TranslatorTest)
     test/logger/translator_test.exs:948
     Assertion with =~ failed
     code:  assert capture_log(:info, fn ->
              ref = Process.monitor(pid)
              Supervisor.start_child(pid, worker(Task, [__MODULE__, :sleep, [self()]]))
              Process.exit(pid, :normal)

              receive do
                {:DOWN, ^ref, _, _, _} -> :ok
              end
            end) =~ ~r"\[info\] Child Task of Supervisor #PID<\d+\.\d+\.\d+> \(Supervisor\.Default\) started
            Pid: #PID<\d+\.\d+\.\d+>
            Start Call: Task.start_link\(Logger.TranslatorTest, :sleep, \[#PID<\d+\.\d+\.\d+>\]\)
            "
     left:  ""
     right: ~r/\[info\] Child Task of Supervisor #PID<\d+\.\d+\.\d+> \(Supervisor\.Default\) started\nPid: #PID<\d+\.\d+\.\d+>\nStart Call: Task.start_link\(Logger.TranslatorTest, :sleep, \[#PID<\d+\.\d+\.\d+>\]\)\n/
     stacktrace:
       test/logger/translator_test.exs:951: (test)

............................

  2) test translates Supervisor progress with name (Logger.TranslatorTest)
     test/logger/translator_test.exs:963
     Assertion with =~ failed
     code:  assert capture_log(:info, fn ->
              ref = Process.monitor(pid)
              Supervisor.start_child(pid, worker(Task, [__MODULE__, :sleep, [self()]]))
              Process.exit(pid, :normal)

              receive do
                {:DOWN, ^ref, _, _, _} -> :ok
              end
            end) =~ ~r"\[info\] Child Task of Supervisor Logger.TranslatorTest started
            "
     left:  ""
     right: ~r/\[info\] Child Task of Supervisor Logger.TranslatorTest started\n/
     stacktrace:
       test/logger/translator_test.exs:966: (test)

..............................................................................................
Finished in 1.0 seconds (0.5s on load, 0.00s async, 0.4s sync)
3 doctests, 157 tests, 2 failures
make: *** [test_logger] Error 2
Build failed, cleaning...
```

Btw. Erlang/OTP 27.0.1 runs this tests just fine.

### Expected behavior

All tests pass without errors.

Btw. Erlang/OTP 27.0.1 runs this tests just fine.

**Comment by josevalim** [comment]
Fixed on main and v1.17 branches already.

**Comment by pertsevds** [comment]
Thank you!
